src/struct.jl

```diff
@@ -54,6 +59,18 @@ end
 
 # We cannot precisely derive the type after `getfield` due to SU limitations,
 # so give up and just say Real.
-SymbolicUtils.promote_symtype(::typeof(getfield), ::Type{<:Struct}, _...) = Real
+function SymbolicUtils.promote_symtype(::typeof(typed_getfield), ::Type{<:Struct{T}}, v::Type{Val{fieldname}}) where {T, fieldname} 
+    FT = fieldtype(T, fieldname)
+    if isstructtype(FT)
+        return Struct{FT}
+    end 
+    FT
+end
+
+
+SymbolicUtils.promote_symtype(::typeof(setfield!), ::Type{<:Struct}, _, ::Type{T}) where T = T
```

Comment:
This now appears twice and is overwritten in line 76, leading to warnings during precompilation.